### PR TITLE
[SPARK-8704] [ML] [PySpark] Add missing methods in StandardScaler 

### PR DESCRIPTION
**Pull request by MechCoder**
Add std, mean to StandardScalerModel
getVectors, findSynonyms to Word2Vec Model
setFeatures and getFeatures to hashingTF

### CODE REVIEW

**Comment by MechCoder** [comment]
This breaks `model.transform(doc).collect()` for `Word2Vec` but I do not understand why.

**Comment by MechCoder** [comment]
It seems that the 'JMap[String, JList[Float]]' is not serializable.

**Comment by MechCoder** [comment]
I pushed in a hack that works, but now the Scala version looks ugly.

**Comment by MechCoder** [comment]
Is there are a way to make a scala method private such that it is accessible only to the python API. It seems `private[python]` works only with classes.

**Comment by JoshRosen** [comment]
Jenkins, retest this please.

**Comment by MechCoder** [comment]
Pushed a commit that will hopefully resolve the pesky Python 3 serialization error. :X . Apparently there is some problem with serialization and deserialization of an Array.

**Comment by mengxr** [comment]
@MechCoder Please split this into multiple PRs:

1. Add methods in Python for those already exist in Scala
2. Add methods in both Scala/Python

This helps code review. Thanks!

**Comment by MechCoder** [comment]
@mengxr 

I just checked that HashingTF inherits from Has NumFeatures so there is no really nothing to do for the first point.

I've just added the methods in scala and python in this PR and it should be open for review now.

**Comment by MechCoder** [comment]
@mengxr updated. I thought it might be slightly overkill to add the scala and python version separately :P

File: mllib/src/main/scala/org/apache/spark/ml/feature/StandardScaler.scala [hunk 1]
```diff
@@ -106,6 +106,12 @@ class StandardScalerModel private[ml] (
     scaler: feature.StandardScalerModel)
   extends Model[StandardScalerModel] with StandardScalerParams {
 
+  /** Standard deviation of the StandardScalerModel*/
+  val std = scaler.std
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/ml/feature/StandardScaler.scala]: need the explicit type info for public methods

**Comment by mengxr** [comment]
LGTM except the Scala style issue.

**Comment by mengxr** [comment]
Merged into master. Thanks!

**Pull request closed without merging**